File: nodejs/device/intel-edison/examples/lib/qiot.js
/**
 * Copyright (c) QNAP Systems, Inc. All rights reserved.
*/

var mqttProtocol = require('./mqtt-connect');
var httpProtocol = require('./http-connect');
var httpsProtocol = require('./https-connect');
var coapProtocol = require('./coap-connect');

const PROTOCOL_TYPE = {
    MQTT: "MQTT",
    HTTP: "HTTP",
    HTTPS: "HTTPS",
    COAP: "COAP"
}

/**
 * define connect protocol
 * @param {(PROTOCOL_TYPE)} type  type of protocol
 */
var createConnection = function(type) {
    var connection;
    switch (type) {
        case PROTOCOL_TYPE.MQTT:
            connection = new mqttProtocol();
            break;
        case PROTOCOL_TYPE.HTTP:
            connection = new httpProtocol();
            break;
        case PROTOCOL_TYPE.HTTPS:
            connection = new httpsProtocol();
            break;
        case PROTOCOL_TYPE.COAP:
            connection = new coapProtocol();
            break;
        default:
            console.log("Not Support protocol");
            break;
    }
    return connection;
}

module.exports = createConnection;
module.exports.protocol = PROTOCOL_TYPE;

File: projects/AWSGreengrass-Integration-Scenarios/QIoT_device_QuAI_Greengrass/RaspberryPi_side/lib/qiot.py
"""
 Copyright (c) QNAP Systems, Inc. All rights reserved.
"""

def connection(type):
    client = None
    if type == protocol.MQTT:
        from lib import mqtt_connect
        client = mqtt_connect.Mqtt()
    elif type == protocol.HTTP:
        from lib import http_connect
        client = http_connect.Http()
    elif type == protocol.HTTPS:
        from lib import https_connect
        client = https_connect.Https()
    elif type == protocol.COAP:
        from lib import coap_connect
        client = coap_connect.Coap()
    return client

class protocol:
    MQTT = "MQTT"
    HTTP = "HTTP"
    HTTPS = "HTTPS"
    COAP = "COAP"
    def __init__(self):
        print 'define protocol'
